packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverActions/useTeachingPopoverActionsStyles.styles.ts

```diff
@@ -1,35 +1,42 @@
-import { makeStyles, mergeClasses } from '@griffel/react';
-import type { SlotClassNames } from '@fluentui/react-utilities';
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
 import type { TeachingPopoverActionsSlots, TeachingPopoverActionsState } from './TeachingPopoverActions.types';
+import type { SlotClassNames } from '@fluentui/react-utilities';
+import { useTeachingPopoverContext_unstable } from '../../TeachingPopoverContext';
 
 export const teachingPopoverActionsClassNames: SlotClassNames<TeachingPopoverActionsSlots> = {
   root: 'fui-TeachingPopoverActions',
-  // TODO: add class names for all slots on TeachingPopoverActionsSlots.
-  // Should be of the form `<slotName>: 'fui-TeachingPopoverActions__<slotName>`
 };
 
-/**
- * Styles for the root slot
- */
 const useStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
+    display: 'flex',
+    flexDirection: 'row',
+    ...shorthands.borderRadius('4px'),
+    columnGap: '8px',
+    rowGap: '8px',
+    paddingTop: '12px',
+  },
+  rootCarousel: {
+    justifyContent: 'space-between',
+  },
+  rootPage: {
+    justifyContent: 'flex-end',
   },
-
-  // TODO add additional classes for different states and/or slots
 });
 
-/**
- * Apply styling to the TeachingPopoverActions slots based on the state
- */
-export const useTeachingPopoverActionsStyles_unstable = (
-  state: TeachingPopoverActionsState,
-): TeachingPopoverActionsState => {
+/** Applies style classnames to slots */
+export const useTeachingPopoverActionsStyles_unstable = (state: TeachingPopoverActionsState) => {
   const styles = useStyles();
-  state.root.className = mergeClasses(teachingPopoverActionsClassNames.root, styles.root, state.root.className);
 
-  // TODO Add class names to slots, for example:
-  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);
+  const totalPages = useTeachingPopoverContext_unstable(context => context.totalPages);
```

Comment:
We should not pull context in style hooks, it should be done under state.

Reply:
@behowell can you follow up and make sure our component creation guidance calls this out or file an issue for it if we need to add a new page about when/where to use component hooks

Reply:
I was doing this in a few places, should all be resolved now.